core/types/block.go

```diff
@@ -421,6 +421,22 @@ func (b *Block) ContainsL1Messages() bool {
 	return false
 }
 
+func (b *Block) NumL1Messages() int {
+	if l1MsgCount := b.l1MsgCount.Load(); l1MsgCount != nil {
```

Comment:
Any concern about reusing this casche in `NumL1MessagesProcessed`?

Reply:
removed the function as it is unused